syntax = "proto3";

service FileControl{
    rpc AddFile(AddFileRequest) returns (AddFileResult);

    rpc DeleteNotExistFile(DeleteFileRequest) returns (DeleteResult);
    rpc DeleteThoseFile(DeleteThoseFileRequest) returns(DeleteThoseFileResult);

    rpc CheckFile(CheckFileRequest) returns(CheckFileResult);
    rpc CheckVolumn(CheckVolumnRequest) returns(CheckVolumnResult);
    rpc CheckDisk(CheckDiskRequest) returns (CheckDiskResult);
    
    rpc LoadFileIntoCache(CachePath) returns(loadCacheResult);
    rpc MoveFile(MoveInfo) returns(MoveResult);
    rpc UnzipFile(UnzipInfo) returns(UnzipResult);
    rpc ZipFile(ZipInfo) returns(ZipResult);
}


message AddFileResult{
    string path=1;
}

message AddFileAnswer{
    enum Result{
        success=0;
        FileNotExist=1;
        DataBaseNotExist=2;
        NotMinimumStorage=3;
        otherFailure=4;
    }
    Result result=1;
    string info=5;
}
